src/vtcomposite.cpp

```diff
@@ -692,104 +716,183 @@ struct LocalizeWorker : Napi::AsyncWorker
+                                    }
+                                    else
+                                    {
+                                        worldviews_to_create = worldviews_for_feature(available_worldviews, baton_data_->worldviews);
+                                        if (worldviews_to_create.empty())
+                                        {
+                                            skip_feature = true;
+                                        }
+                                    }
+                                }
+                                else
+                                {
+                                    skip_feature = true;
+                                }
+                            }
+                            else // safeguard – should never reach here
+                            {
+                                skip_feature = true;
+                            }
                         }
 
-                        if (property_key == baton_data_->class_property)
+                        else if (
+                            utils::startswith(property_key, baton_data_->class_property) ||
+                            utils::startswith(property_key, baton_data_->class_prefix + baton_data_->class_property))
                         {
-                            if (property.value().type() == vtzero::property_value_type::string_value)
+                            // check if the property is of higher precedence that class key encountered so far
+                            std::uint32_t idx = static_cast<std::uint32_t>(std::find(class_key_precedence.begin(), class_key_precedence.end(), property_key) - class_key_precedence.begin());
+                            if (idx < class_key_idx)
                             {
+                                class_key_idx = idx;
                                 class_value = property.value();
                             }
-                            continue;
+                            // wait till we are done looping through all properties before we add class value to final_properties
                         }
 
-                        // preserve original params.language_property value
-                        if (property_key == baton_data_->language_property)
+                        else if (
+                            utils::startswith(property_key, baton_data_->language_property) ||
+                            utils::startswith(property_key, baton_data_->language_prefix + baton_data_->language_property))
                         {
-                            name_value = property.value();
+                            // check if the property is of higher precedence that class key encountered so far
+                            std::uint32_t idx = static_cast<std::uint32_t>(std::find(language_key_precedence.begin(), language_key_precedence.end(), property_key) - language_key_precedence.begin());
```

Comment:
also here would std::distance be better?

Reply:
Changed to std::distance in 3711e05918f95ff3494302a0c36e07fe1bdfad08.